### DISCUSSION TITLE
Read data by column

### DISCUSSION THREAD

**Discussion by SreekanthMekala** [Claviz/xlstream#36] [Q&A] [answered]
We are trying to read data by specific column, Can you please give us details on this. TIA

**Comment by AlexTimo** [comment]
Hi!

Could you please attach a sample Excel file and describe what you'd like to achieve?

**Comment by SreekanthMekala** [comment]
[sample_column.xlsx](https://github.com/Claviz/xlstream/files/6501168/sample_column.xlsx)

@AlexTimo , We have this sample need to get unique data for column B and C. If small file we can get all data by rows and filter it, but some files like it has 50 columns and 100K rows that time its taking huge time to read file. It will be help that we can read by column so we can make read data faster. Please let us know if any solutions. TIA

**Reply by AlexTimo** [reply]
Hi @SreekanthMekala, 

do you have any further question or should I close this thread?

**Reply by SreekanthMekala** [reply]
Thanks for update @AlexTimo. We can close it

**Comment by AlexTimo** [comment, marked as answer]
I've a done a test with 100K rows and 50 columns, it took about 2 min and 20 seconds to parse the file and find unique values in two columns. Since xlstream is optimized for maintaining low memory footprint, you may be better served by a library that loads the whole file into memory first. For example, the same task (finding unique values in two columns) took about 50 seconds when using SheetJS (xlsx package).
